drivers/serial/serial_xen.c

```diff
@@ -40,26 +42,29 @@ int xen_serial_setbrg(struct udevice *dev, int baudrate)
 static int xen_serial_probe(struct udevice *dev)
 {
 	struct xen_uart_priv *priv = dev_get_priv(dev);
-	u64 v = 0;
+	u64 val = 0;
 	unsigned long gfn;
-	int r;
+	int ret;
 
-	r = hvm_get_parameter(HVM_PARAM_CONSOLE_EVTCHN, &v);
-	if (r < 0 || v == 0)
-		return r;
+	ret = hvm_get_parameter(HVM_PARAM_CONSOLE_EVTCHN, &val);
+	if (ret < 0 || val == 0)
+		return ret;
```

Comment:
can we return ret? What does hvm_get_parameter return? Is it a POSIX errno, is it aligned with the definitions used by U-boot?
If ret >=0 and val == 0 - this is still an error condition. And you return ret which might be zero letting the caller know that everything is good and effectively hiding the error case

Reply:
it seems hvm_get_parameter returns a POSIX errno, so it should be safe